cvat/apps/engine/frame_provider.py

```diff
@@ -429,8 +436,24 @@ def _get_segment(self, validated_frame_number: int) -> models.Segment:
 
         return segment
 
+    def unload(self):
+        self._clear_segment_frame_provider_cache()
```

Comment:
This is a reasonable implementation, but why not also call `unload` on every provider in the cache?

Reply:
Because it can unload the cached providers that are referenced by some other code. If they are not referenced, they are cleaned anyway.